lib/ddtrace/contrib/configuration/integration.rb

```diff
@@ -0,0 +1,79 @@
+module Datadog
+  module Contrib
+    module Configuration
+      # Represents an instance of a configurable integration
+      class Integration
```

Comment:
Mulling a rename from `Integration` to `Component`, to avoid confusion with other integration code; any thoughts?

Reply:
I'm trying to think if I'd see this concept being used for "components" that are also **not** "top-level" integration. In other words, something that can only be a sub-component of a top-level integration, but never used standalone. If this scenario can happen, then `Integration` is a conflicting name here.

I'm leaning more towards `Component` right now: and integration is always a components, but a component doesn't have to commit to being a full integration.

Reply:
Following up the previous comment, I think that most of our configuration and activation behaviour would be responsibility of the "component", and top-level "integrations" would have additional logic on how to properly activate its components.

Reply:
So it's interesting that you were comparing the idea of "top-level" to "sub-level"; I had asked this question as well, because I had considered maybe our `Datadog.configuration` object could simply implement this exact same module, which would make the two identical.

After considering though, they are not identical. Although all integrations can be activated independently of a parent integration (e.g. ActiveSupport without Rails), the process by which "top-level" and "sub-level" are activated is not the same.

"Top-level" configures and activates immediately, "sub-level" stores config and defers activation to the parent integration that manages it (e.g. Rails.) Then later, that integration can activate it as it sees appropriate.

Therefore we need some slightly different functions & interfaces to invoke these behaviors, unless we were to rethink `#use` to incorporate in this "deferment" feature.

Reply:
I think we are in good agreement here.
And regarding rethinking `#use` and deferment, I'm ok with this being addressed in a future PR.